### PR TITLE
Add IoPoolSize config

### LINKED ISSUE #23349: [Enhancement]: Need configuration items to control the resource usage while loading
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

QueryNodeV2 performs loading segments more radically, also consumes more memory, we need parameters to control it

### Why is this needed?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
resolve #23349

### CODE REVIEW

File: internal/querynodev2/segments/segment_loader.go [hunk 1]
```diff
@@ -70,10 +70,42 @@ type Loader interface {
...
+	// make sure small machines could load faster
+	if ioPoolSize < 32 {
+		ioPoolSize = 32
+	}
+	// limit the number of concurrency
+	if ioPoolSize > 256 {
+		ioPoolSize = 256
+	}
+
+	if configPoolSize := paramtable.Get().QueryNodeCfg.IoPoolSize.GetAsInt(); configPoolSize > 0 {
```
Comment (congqixia) [inline internal/querynodev2/segments/segment_loader.go:88]: does `at most 256` rules apply to config value?

Comment (congqixia) [reply internal/querynodev2/segments/segment_loader.go:88]: And `at least 32` rule?

Comment (yah01) [reply internal/querynodev2/segments/segment_loader.go:88]: No

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**